### DISCUSSION TITLE
Running from a different drive (Host OS)

### DISCUSSION THREAD

**Discussion by Joanone** [neurodesk/neurodesk.github.io#116] [Q&A] [answered]
Dear Neurodesk experts,

Joan here. I just started using this virtual container recently and its really brilliant. Thanks.

However, i have some questions:

1) Do i need to run the .exe file everytime i want to access neurodesk subsequently?

2) Can i run it from a different drive on my host OS? because i realized that my C-drive seemed to fill up so fast during a sample analysis i performed using the neurodesk.

3) Is 32GB RAM size okay for computationally extensive analysis with softwares like Mrtrix on the neurodesk?

Thanks for your prompt responses.

**Comment by civier** [comment]
Dear Joan,

We are happy that you like Neurodesk. Hope you enjoy using it.

Regarding 1 and 2: @stebo85 @aswinnarayanan, can you please assign this to somebody from the team who can test Neurodesk on Window (I can test Neurodesk on Mac and Linux only).

Regarding 3: 32 GB should be more than enough for most analyses. However, if you're running Fixel-based analysis (FBA), it is possible that more than 32GB RAM will be required (see point 7 here: https://mrtrix.readthedocs.io/en/0.3.16/workflows/fixel_based_analysis.html)
Also, the dwifslpreproc command (https://mrtrix.readthedocs.io/en/latest/reference/commands/dwifslpreproc.html) uses FSL's topup under the hood, and from my experience, it might have issues with memory, especially when being ran in a container (not our fault; topup is just not well implemented). If you encounter any issues, please let us know. I'm also contributing to MRtrix, so should be able to help.

Good luck,
Oren

P.S.
Would you mind to share with us your affiliation? (only if you want) We are always interested to know where our users are from, as we really seek a worldwide reach.

**Comment by Joanone** [comment]
Hello Oren,

Thanks for your prompt response.

I look forward to receiving feedbacks concerning 1 and 2.

Thanks for your response regarding 3. No, I won’t be running FBA, I’m just interested in building the structural connectome, so I’m following the steps documented here: https://mrtrix.readthedocs.io/en/latest/quantitative_structural_connectivity/ismrm_hcp_tutorial.html

I’m also using the HCP data as stated in the tutorial link above, however I realized the analysis froze when it got to the connectome generation, step 1 on neurodesk. I think 100m tracks might be a lot, are there resources you could link me to which suggests reducing the number of tracks?  Or is just advisable to increase the size of my RAM rather? I’m currently on 16GB.

Based on what I’ve read so far, I’m not sure I would need the topup as it seems my data is already preprocessed. I’m also open to correction on this.

Sure, I don’t mind sharing. I’m a PhD research student at the University of Electronic Science and Technology, China.

Kind regards,
Joan

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

**Comment by civier** [comment]
Hello Joan,

My pleasure.

Constructing a structural connectome should not be limited by memory size, so I'm not sure why the analysis got stuck. Did you check that you have enough disk space? (100M streamlines will require around 100G diskspace) If it's not an issue of storage, can you post here a screenshot of when it stopped working?

It's really off-topic for this forum, but the instructions that you're using are really outdated. You are welcome to follow my instructions instead if you wish. They are here: https://github.com/civier/HCP-dMRI-connectome

Good luck,
Oren

**Comment by Joanone** [comment]
Hi Oren,

Thanks once again!

Yes, I do agree that it was an issue of storage. My disk space was less than 100G before I started running the analysis. I’ll create room for that and try it again.

I really do appreciate the instructions you shared, I’ll be following them and keep you posted when I have any questions.

Kind regards,
Joan

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

**Comment by stebo85** [comment, marked as answer]
Dear @Joanone,

> Do i need to run the .exe file everytime i want to access neurodesk subsequently?

@thomshaw92 developed the exe file for neurodesk to make it easier for people to get started. He can comment more, but yes - if you want to run Neurodesk you need to start this file everytime you use it.

There is also the manual way of starting neurodesk, which is also described in the documentation:  https://neurodesk.github.io/docs/neurodesktop/getting-started/windows/#option-2-using-terminal

Open a powershell terminal and run:
`docker run --shm-size=1gb -it --privileged --name neurodesktop -v C:/neurodesktop-storage:/neurodesktop-storage -p 8080:8080 -h neurodesktop-20220111 vnmd/neurodesktop:20220111`



> Can i run it from a different drive on my host OS? because i realized that my C-drive seemed to fill up so fast during a sample analysis i performed using the neurodesk.

Yes, this is possible, although the .exe version doesn't support this yet as far as I know. To mount other storage you need to add another -v flag to the docker command (described here: https://neurodesk.github.io/docs/neurodesktop/storage/#mounting-network-storage-on-your-host-computer), e.g.: 
`docker run --shm-size=1gb -it --privileged --name neurodesktop -v C:/neurodesktop-storage:/neurodesktop-storage -v D:/moredata:/data -p 8080:8080 -h neurodesktop-20220111 vnmd/neurodesktop:20220111`



> Is 32GB RAM size okay for computationally extensive analysis with softwares like Mrtrix on the neurodesk?

This will depend on the application you are running, but should generally get you quite far :)

**Reply by Joanone** [reply]
Thanks so much @stebo85 . Your responses are very helpful.'

Kind regards.

**Reply by Joanone** [reply]
Hello @stebo85 , Please I have a follow up question concerning mounting another storage as you previously mentioned

docker run --shm-size=1gb -it --privileged --name neurodesktop -v C:/neurodesktop-storage:/neurodesktop-storage -v D:/moredata:/data -p 8080:8080 -h neurodesktop-20220111 vnmd/neurodesktop:20220111

I ran this command above in the power shell but i got this error. what do you think i might be missing?

![image](https://user-images.githubusercontent.com/83198624/151312225-883db350-184e-4aa6-8177-0e30f133157d.png)

**Comment by civier** [comment]
Hello @Joanone

It is all fine. It only means that you didn't delete the previous Neurodesktop that was running. Please follow the instructions here:
https://neurodesk.github.io/docs/neurodesktop/getting-started/windows/#stopping-neurodesktop
and then run the command to start neurodesktop again.
(just remember that anything that was saved outside of /neurodesktop-storage in the previous neurodesktop would be deleted, so make sure to move all your files to /neurodesktop-storage before deleting the previous Neurodesktop)

Good luck,
Oren

**Reply by Joanone** [reply]
okay. Thanks very much @civier. I totally understand now. i'll follow suit.
